ecl/data_source_ecl_network_fic_gateway_v2.go

```diff
@@ -0,0 +1,128 @@
+package ecl
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/nttcom/eclcloud/ecl/network/v2/fic_gateways"
+)
+
+func dataSourceNetworkFICGatewayV2() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNetworkFICGatewayV2Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"fic_service_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"fic_gateway_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"qos_option_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"tenant_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceNetworkFICGatewayV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	networkClient, err := config.networkV2Client(GetRegion(d, config))
+
+	listOpts := fic_gateways.ListOpts{}
+
+	if v, ok := d.GetOk("description"); ok {
+		listOpts.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("fic_service_id"); ok {
+		listOpts.FICServiceID = v.(string)
+	}
+
+	if v, ok := d.GetOk("fic_gateway_id"); ok {
+		listOpts.ID = v.(string)
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		listOpts.Name = v.(string)
+	}
+
+	if v, ok := d.GetOk("qos_option_id"); ok {
+		listOpts.QoSOptionID = v.(string)
+	}
+
+	if v, ok := d.GetOk("status"); ok {
+		listOpts.Status = v.(string)
+	}
+
+	if v, ok := d.GetOk("tenant_id"); ok {
+		listOpts.TenantID = v.(string)
+	}
+
+	pages, err := fic_gateways.List(networkClient, listOpts).AllPages()
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve fic_gateways: %s", err)
```

Comment:
エラー文字列は大文字で始めないでください。
https://github.com/golang/go/wiki/CodeReviewComments#error-strings
エラーを出力する場合は`%w`を使用してください。
https://blog.golang.org/go1.13-errors
```suggestion
		return fmt.Errorf("unable to retrieve fic_gateways: %w", err)
```